internal/test/builder/infrastructure.go

```diff
@@ -35,6 +35,11 @@ var (
 	// GenericInfrastructureMachineTemplateCRD is a generic infrastructure machine template CRD.
 	GenericInfrastructureMachineTemplateCRD = untypedCRD(InfrastructureGroupVersion.WithKind(GenericInfrastructureMachineTemplateKind))
 
+	// GenericInfrastructureMachinePoolTemplateKind is the Kind for the GenericInfrastructureMachinePoolTemplate.
+	GenericInfrastructureMachinePoolTemplateKind = "GenericInfrastructureMachineTemplate"
```

Comment:
I'm not sure how you are planning to use them, but it would be good to have:
* GenericInfrastructureMachinePool
* GenericInfrastructureMachinePoolTemplate

(with all the const, godoc and const values correct :))

Reply:
Let's also add the corresponding TestInfrastructureMachinePoolKind and TestInfrastructureMachinePoolTemplateKind stuff below

Reply:
I prematurely copied those from https://github.com/kubernetes-sigs/cluster-api/compare/main...willie-yao:cluster-api:cc-mp-unit-tests#diff-2f9020e9e700d4c75c18fd48939938fde0ffa677d28886bf139d4cd78c87175aR38. But it looks like they are not being referenced there as well. It would suggest to drop those from this PR again and to not add the `TestInfrastructureMachinePool` consts and defer that to the point when they are actually needed. WDYT?

Reply:
Sounds fine to me